### PR TITLE
missing helper file error

### PR DESCRIPTION
**Pull request by mel-d**

### CODE REVIEW

**Comment by mel-d** [comment]
Okay, I'm going to attempt to explain my error because I think I know what's causing it but not how to fix it.
`/Users/melindadeinert/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/actionpack-4.2.6/lib/abstract_controller/helpers.rb:151:in `rescue in block in modules_for_helpers': Missing helper file helpers//users/melindadeinert/code/tag_awards/app/helpers/application_helper.rb_helper.rb (AbstractController::Helpers::MissingHelperError)`

So there are a bunch of Stack Overflow articles (like this one http://stackoverflow.com/questions/27871726/strange-error-in-rails-missing-helper/27966165#27966165) from people who say there's some kind of bug in this version of Ruby re: the capitalization of file paths, and they were able to fix this error by changing the capitalization of the file path to match the error.

But as far as I can tell, the only difference in capitalization between the file path in the error and my local files is that I have Code capitalized on my computer. I don't want to change that, do I?

**Comment by mel-d** [comment]
The only other thing that has crossed my mind is that I don't have any mention of rb_helper.rb within my application_helper.rb file. Maybe I need to require it or add something?

File: app/models/user.rb [hunk 1]
```diff
@@ -22,6 +22,11 @@ def delete_previous_noms(user_id)
     end
   end
 
+  #determines whether a specific nomination object has been saved previously
```
Comment (sumeetjain) [inline app/models/user.rb:25]: Missing space after #.

File: app/models/user.rb [hunk 2]
```diff
@@ -22,6 +22,11 @@ def delete_previous_noms(user_id)
...
+  #
+  #takes in Integer of award_id, Integer of nom_count, and String of value_needed
```
Comment (sumeetjain) [inline app/models/user.rb:27]: Missing space after #.

Comment (sumeetjain) [inline app/models/user.rb:27]: Line is too long. [81/80]

File: app/models/user.rb [hunk 3]
```diff
@@ -22,6 +22,11 @@ def delete_previous_noms(user_id)
...
+  #
+  #returns String of existing nomination or empty String if no previous nomination
```
Comment (sumeetjain) [inline app/models/user.rb:29]: Missing space after #.

Comment (sumeetjain) [inline app/models/user.rb:29]: Line is too long. [82/80]

File: app/models/user.rb [hunk 4]
```diff
@@ -33,6 +38,11 @@ def nomination_value(award_id, nom_count, value_needed)
     end     
   end  
 
+  #saves nominations entered by user
```
Comment (sumeetjain) [inline app/models/user.rb:41]: Missing space after #.

File: app/models/user.rb [hunk 5]
```diff
@@ -33,6 +38,11 @@ def nomination_value(award_id, nom_count, value_needed)
...
+  #
+  #takes in Integer of user_id and Hash of awards params from nominations ballot
```
Comment (sumeetjain) [inline app/models/user.rb:43]: Missing space after #.

File: app/models/user.rb [hunk 6]
```diff
@@ -33,6 +38,11 @@ def nomination_value(award_id, nom_count, value_needed)
...
+  #
+  #returns nil
```
Comment (sumeetjain) [inline app/models/user.rb:45]: Missing space after #.

File: app/models/user.rb [hunk 7]
```diff
@@ -60,10 +70,16 @@ def record_nominations(user_id, nominations_hash)
   
   private
 
+  #assigns random key to a user
```
Comment (sumeetjain) [inline app/models/user.rb:81]: Missing space after #.

File: app/models/user.rb [hunk 8]
```diff
@@ -60,10 +70,16 @@ def record_nominations(user_id, nominations_hash)
...
+  #
+  #uses SecureRandom gem
```
Comment (sumeetjain) [inline app/models/user.rb:83]: Missing space after #.

File: app/models/user.rb [hunk 9]
```diff
@@ -60,10 +70,16 @@ def record_nominations(user_id, nominations_hash)
...
   def set_voter_token
     self.voter_token = generate_token
   end
 
+  #generates a random 6-digit alphanumeric key
```
Comment (sumeetjain) [inline app/models/user.rb:88]: Missing space after #.

File: app/models/user.rb [hunk 10]
```diff
@@ -60,10 +70,16 @@ def record_nominations(user_id, nominations_hash)
...
+  #
+  #uses SecureRandom gem
```
Comment (sumeetjain) [inline app/models/user.rb:90]: Missing space after #.

File: app/models/user.rb [hunk 11]
```diff
@@ -43,27 +52,42 @@ def record_nominations(user_id, nominations_hash)
...
       end
     end
   end
+
+  #saves object of nomination entered by user
+  #
+  #takes in Integer of user_id, Integer of award id, and Collection of Hashes from nested nominations_hash
+  #
+  #returns nil
+  def save_nomination_object(user_id, current_award, value3)
```
Comment (sumeetjain) [inline app/models/user.rb:67]: Has approx 9 statements (TooManyStatements)

Comment (sumeetjain) [inline app/models/user.rb:67]: Has the parameter name 'value3' (UncommunicativeParameterName)

File: app/models/user.rb [hunk 12]
```diff
@@ -43,27 +52,42 @@ def record_nominations(user_id, nominations_hash)
...
-            @new_nom = Nomination.new
-            @new_nom.user_id = user_id
-            @new_nom.award_id = @current_award
-            @new_nom.open = true
-            @new_nom.theater = value3["theater"]
-            @new_nom.nominee = value3["nominee"]
-            @new_nom.role = value3["role"]
-            @new_nom.show = value3["show"]
-            @new_nom.save
+            save_nomination_object(user_id, @current_award, value3)
```
Comment (sumeetjain) [inline app/models/user.rb:55]: Place the condition on the same line as `elsif`.

File: app/models/user.rb [hunk 13]
```diff
@@ -43,27 +52,42 @@ def record_nominations(user_id, nominations_hash)
...
           end
         end
       end
     end
   end
+
+  #saves object of nomination entered by user
```
Comment (sumeetjain) [inline app/models/user.rb:62]: Missing space after #.

File: app/models/user.rb [hunk 14]
```diff
@@ -43,27 +52,42 @@ def record_nominations(user_id, nominations_hash)
...
+  #
+  #takes in Integer of user_id, Integer of award id, and Collection of Hashes from nested nominations_hash
```
Comment (sumeetjain) [inline app/models/user.rb:64]: Missing space after #.

Comment (sumeetjain) [inline app/models/user.rb:64]: Line is too long. [106/80]

File: app/models/user.rb [hunk 15]
```diff
@@ -43,27 +52,42 @@ def record_nominations(user_id, nominations_hash)
...
+  #
+  #returns nil
```
Comment (sumeetjain) [inline app/models/user.rb]: Missing space after #.

File: app/models/user.rb (hunk 11 above)
Comment (sumeetjain) [inline app/models/user.rb:67]: Inconsistent indentation detected.

**Pull request merged**